hrjobcontract/tests/phpunit/CRM/Hrjobcontract/BAO/HRJobContractRevisionTest.php

```diff
@@ -1,32 +1,26 @@
 <?php
 
-require_once 'CiviTest/CiviUnitTestCase.php';
+use Civi\Test\HeadlessInterface;
+use Civi\Test\TransactionalInterface;
 
 /**
  * Class CRM_Hrjobcontract_BAO_HRJobContractRevisionTest
+ *
+ * @group headless
  */
-class CRM_Hrjobcontract_BAO_HRJobContractRevisionTest extends CiviUnitTestCase {
+class CRM_Hrjobcontract_BAO_HRJobContractRevisionTest extends PHPUnit_Framework_TestCase implements
+  HeadlessInterface,
+  TransactionalInterface {
+
+  public function setUpHeadless() {
+    return \Civi\Test::headless()->installMe(__DIR__)->apply();
+  }
+
   function setUp() {
-    parent::setUp();
-    $this->quickCleanup(array(
-      'civicrm_hrjobcontract_details',
-      'civicrm_hrjobcontract_health',
-      'civicrm_hrjobcontract_role',
-      'civicrm_hrjobcontract_hour',
-      'civicrm_hrjobcontract_pay',
-      'civicrm_hrjobcontract_leave',
-      'civicrm_hrjobcontract_pension',
-      'civicrm_hrjobcontract_revision',
-      'civicrm_hrjobcontract',
-    ));
     $upgrader = CRM_Hrjobcontract_Upgrader::instance();
     $upgrader->install();
```

Comment:
You're right here. The upgrade actually runs during the extension installation. I did a quick test here by removing the manual upgrader call on EntitlementCalculationTest and the tests still pass. I probably solved the problem with something else and thought it was fixed by the manual call :(